R/plot_qq.R

```diff
@@ -0,0 +1,23 @@
+#' Plot the quantile-quantile for `fit` with a 1:1 line
+#'
+#' @param data A list created by [sdmTMB::sdmTMB()].
+#' @template dir
+#'
+#' @author Chantel R. Wetzel
+#' @export
+plot_qq <- function(data, dir) {
+  residuals <- data[["data"]][["residuals"]]
+
+	grDevices::png(
+		filename = file.path(dir, "qq.png"),
+    width = 7,
+    height = 7,
+    units = "in",
+    res = 300,
+    pointsize = 12
+  )
+  on.exit(dev.off())
+  stats::qqnorm(residuals)
```

Comment:
I'm not sure what the evolving best practice is here. These (randomized quantal residuals with the Laplace approximation) and DHARMa residuals with the Laplace approximation can definitely look off even if the model is consistent with the data. MCMC residuals with fixed effects at their MLE seem to work well but take longer. That's what they used [here](https://doi.org/10.1002/eap.2453). `?residuals.sdmTMB` https://pbs-assess.github.io/sdmTMB/articles/web_only/residual-checking.html 

Note that because of esoteric problems on CRAN, we had to split that out into its [own (non-CRAN) package](https://github.com/pbs-assess/sdmTMBextra).

```r
samps <- sdmTMBextra::predict_mle_mcmc(fit, mcmc_iter = 201, mcmc_warmup = 200)
mcmc_res <- residuals(fit_nb2, type = "mle-mcmc", mcmc_samples = samps)
qqnorm(mcmc_res);qqline(mcmc_res)
```

In a perfect world you'd sample with more MCMC iterations first.

Reply:
I am going to make this a separate issue. Thanks so much @seananderson for giving the suggestion and providing code to do it.